sdk/scheduler/src/main/java/com/mesosphere/sdk/framework/FrameworkConfig.java

```diff
@@ -132,6 +144,20 @@ public static FrameworkConfig fromEnvStore(EnvStore envStore) {
         envStore.getOptionalNonEmpty("FRAMEWORK_WEB_URL", ""));
   }
 
+  /**
+   * Returns the configured Mesos role to use for running the service.
+   * Priority is given to the role specified by MESOS_ALLOCATION_ROLE environment variable set by Mesos.
+   * If no such environment-variable is found, or MARATHON_APP_ENFORCE_GROUP_ROLE is false we revert to the legacy
+   * version which uses the service name. Leading slashes are removed and intermediate slashes are escaped.
+   */
+  private static String getServiceRole(String frameworkName, Optional<String> serviceNamespace) {
+    if (serviceNamespace.isPresent()) {
```

Comment:
(nit) simplify :
```java
    return serviceNamespace.map(SchedulerUtils::withEscapedSlashes).orElseGet(() -> getServiceRole(frameworkName));
```

Reply:
Let's merge this method with the ` private static String getServiceRole(String frameworkName)` (https://github.com/mesosphere/dcos-commons/pull/3102/files#diff-c06ef9fa9ce86e5f8888d227fa91d0cfR177) . Even the javadoc needs to be merged/rephrased. Both are `private static` so it should be straight forward.